.buildkite/bindists-pipeline

```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+BAZEL_DIR="$(.buildkite/fetch-bazel-bindist)"
+trap "rm -rf '$BAZEL_DIR'" EXIT
+export PATH="$BAZEL_DIR:$PATH"
+echo "common:ci --build_tag_filters -requires_lz4,-requires_proto,-requires_zlib,-requires_doctest,-requires_c2hs,-requires_threaded_rts,-dont_test_with_bindist" > .bazelrc.local
+# XXX: See .bazelrc [backward compatible options] for the the rational behind this flag
+echo "build --incompatible_use_python_toolchains=false" >> .bazelrc.local
+# XXX: Remove once all rules using `use_default_shell_env = True` have been
+#   converted to using `rules_sh`'s POSIX toolchain.
+echo "build:ci --experimental_strict_action_env" >> .bazelrc.local
+./tests/run-start-script.sh --use-bindists
+bazel build --config ci //tests/...
```

Comment:
If the environment changes propagate to the next buildkite step, we probably should move this to `pipeline.yml`, like in the nixpkgs case.

Reply:
I tried that initially and found that
1) You have to be careful to escape variables. E.g. `\$PATH` instead of `$PATH`.
2) `export PATH=...` didn't seem to be picked up and `bazel` called the globally installed version instead.

So, putting the steps into a dedicated script made things much easier and less error prone.

FWIW the [examples](https://github.com/buildkite/bash-example/blob/aa4f98e314c0350edc6b5bd6858c072ccc93f2ac/.buildkite/pipeline.yml#L3) recommend the same thing.

Reply:
I'm mostly concerned about maintaining those three lines:
```
./tests/run-start-script.sh --use-(bindists,nix)
bazel build --config ci //tests:run-tests
./bazel-ci-bin/tests/run-tests
```

in two places.

What about moving that into a `build.sh` script, which assumes a right `bazel` to be in `$PATH`, and calling it from both pipelines?

We could have `build.sh` accepting the `--use-?` as a parameter.

Reply:
Those lines are different between `nixpkgs` and bindists.

In the `nixpkgs` pipeline they are:
```
      ./tests/run-start-script.sh --use-nix
      bazel build --config ci //tests:run-tests
      ./bazel-ci-bin/tests/run-tests
      bazel coverage //tests/... --config ci --build_tag_filters "coverage-compatible" --test_tag_filters "coverage-compatible" --test_output=all
```

And in the bindists pipeline they are:
```
./tests/run-start-script.sh --use-bindists
bazel build --config ci //tests/...
```

Once we've fixed what's missing for the bindist so it can run the same steps as nixpkgs then we should indeed unify these steps into something like `build.sh`.

Reply:
Can you create a followup issue for that?